packages/manager/src/managers/git/GitManager.ts

```diff
@@ -464,6 +465,23 @@ export class GitManager extends BaseManager {
 		}
 	}
 
+	async detectGitProvider(): Promise<string> {
+		try {
+			const remoteUrl = execaCommandSync("git remote get-url origin");
+			const domainRegex = /(?:https?:\/\/|git@)([^:/]+)[/:]/i;
+			const match = remoteUrl.stdout.match(domainRegex);
+			const domain = match && match.length > 0 ? match[1] : "";
+
+			return domain;
```

Comment:
We use `"gitHub" | "gitLab" | "bitbucket"` as provider identifiers, which are the camelcase versions of each provider's name.

See https://github.com/prismicio/slice-machine/blob/9388a4b033a69b61abb4b57e6831abf722237519/packages/manager/src/constants/GIT_PROVIDER.ts#L5. We only support GitHub today, but we plan to support GitLab and Bitbucket in the future.

We'll probably want consistency in a general purpose method like this one. Is it possible for us to map the values for providers we know about?

Reply:
I did that at the beginning, but it was less powerful than this. The goal here is really to just get the domain and let the domain tell us what they us. 

Thanks to this and not having a fixed list, it will help to get all possible providers / custom self hosted solution. I would differentiate the tracking compare to the git integration then. WDYT?

I will merge that but we can always change after without a problem!

Reply:
When we need to use `detectGitProvider` for other uses, we can revisit. This version works well for tracking, indeed. 🙂